packages/squarev2-integration/src/squarev2-payment-processor.spec.ts

```diff
@@ -187,6 +187,34 @@ describe('SquareV2PaymentProcessor', () => {
             expect(onValidationChange).toHaveBeenNthCalledWith(2, true);
             expect(onValidationChange).toHaveBeenNthCalledWith(3, false);
         });
+
+        it("should exempt postalCode from validation if it hasn't been rendered yet", async () => {
+            const onValidationChange = jest.fn();
+
+            await processor.initializeCard({
+                containerId: 'card-container',
+                onValidationChange,
+            });
+            squareV2MockFunctions.simulateEvent(
+                'focusClassAdded' as CardInputEventTypes,
+                'cardNumber' as CardFieldNames,
+                true,
+            );
+            squareV2MockFunctions.simulateEvent(
+                'focusClassRemoved' as CardInputEventTypes,
+                'expirationDate' as CardFieldNames,
+                true,
+            );
+            squareV2MockFunctions.simulateEvent(
+                'errorClassRemoved' as CardInputEventTypes,
+                'cvv' as CardFieldNames,
+                true,
+            );
+
+            expect(onValidationChange).toHaveBeenCalledTimes(2);
```

Comment:
It seems like we have 3 events being fired, and only two times `onValidationChange` is triggered. Can you please explain that?

Reply:
It's because of the [distinctuntilchanged](https://www.learnrxjs.io/learn-rxjs/operators/filtering/distinctuntilchanged) operator.
> Only emit when the current value is different than the last.

The observer in this case `onValidationChange` will only receive a value different than the last. So intead of having something like this:
onValidationChange(false);

Reply:
Makes sense, so it will fire once with false regardless. Thanks for explanation.